Source/JavaScriptCore/parser/Parser.cpp

```diff
@@ -3289,7 +3333,12 @@ template <class TreeBuilder> TreeSourceElements Parser<LexerType>::parseClassFie
         }
 
         // Only valid class fields are handled in this function.
-        ASSERT(match(EQUAL) || match(SEMICOLON) || match(CLOSEBRACE) || m_lexer->hasLineTerminatorBeforeToken());
+        ASSERT(match(EQUAL) || match(SEMICOLON) || match(CLOSEBRACE) || match(OPENBRACE) || m_lexer->hasLineTerminatorBeforeToken());
+
+        if (match(OPENBRACE) && isStaticField) {
+            context.appendStatement(sourceElements, parseBlockStatement(context));
+            continue; 
+        }
```

Comment:
Here, I just parse the static block again which is redundant. How can we get the corresponding Class node or Static block node?